src/community.c

```diff
@@ -2098,6 +2098,223 @@ int igraph_le_community_to_membership(const igraph_matrix_t *merges,
 
 /********************************************************************/
 
+/**
+ * \ingroup communities
+ * \function igraph_community_fluid_communities
+ * \brief Community detection algorithm based on the simple idea of  
+ * several fluids interacting in a non-homogeneous environment
+ * (the graph topology), expanding and contracting based on their
+ * interaction and density. 
+ *
+ * This function implements the community detection method described in:
+ * Parés F., Garcia-Gasulla D., et al.: Fluid Communities: A Community
+ * Detection Algorithm. [https://arxiv.org/abs/1703.09307].
+ *
+ * \param graph The input graph. Only undirected, unweighted graphs are
+ * supported.
+ * \param no_of_communities The number of communities to be found.
+ * \param membership The result vector mapping vertices to the communities
+ * they are assigned to.
+ * \param modularity If not a null pointer, then it must be a pointer
+ *   to a real number. The modularity score of the detected community
+ *   structure is stored here.
+ * \return Error code. 
+ * 
+ * Time complexity: O(|E|)
+ * 
+ * \example examples/simple/igraph_community_fluid_communities.c
+ */
+int igraph_community_fluid_communities(const igraph_t *graph,
+                                        int *no_of_communities,
+                                        igraph_vector_t *membership,
+                                        igraph_real_t *modularity) {
+  long int no_of_nodes = igraph_vcount(graph);
+  long int i, j, k, kv1;
+  igraph_adjlist_t al;
+  double max_density = 1.0;
+  igraph_bool_t running = 1;
+
+  igraph_vector_t node_order, density, label_counters, dominant_labels, nonzero_labels;
+  igraph_vector_int_t com_to_numvertices;
+
+  /* Resize membership vector (number of nodes) */
+  IGRAPH_CHECK(igraph_vector_resize(membership, no_of_nodes));
+
+  /* Initialize density and com_to_numvertices vectors */
+  IGRAPH_CHECK(igraph_vector_init(&density, (long int) *no_of_communities));
+  IGRAPH_FINALLY(igraph_vector_destroy, &density);
+  IGRAPH_CHECK(igraph_vector_init(&com_to_numvertices, (long int) *no_of_communities));
```

Comment:
`igraph_vector_init()` is being used to initialize an `igraph_vector_int_t`. Use `igraph_vector_int_init()`.